core/src/Lender.sol

```diff
@@ -104,26 +104,28 @@ contract Lender is Ledger {
     //////////////////////////////////////////////////////////////*/
 
     function deposit(uint256 amount, address beneficiary) public returns (uint256 shares) {
-        // Guard against reentrancy, accrue interest, and update reserves
+        // Accrue interest and update reserves
         (Cache memory cache, uint256 inventory) = _load();
 
+        // Convert `amount` to `shares`
         shares = _convertToShares(amount, inventory, cache.totalSupply, /* roundUp: */ false);
         require(shares != 0, "Aloe: zero impact");
 
+        // Mint shares, track rewards, and (if applicable) handle courier accounting
+        cache.totalSupply = _mint(beneficiary, shares, amount, cache.totalSupply);
+        // Assume tokens are transferred
+        cache.lastBalance += amount;
+
+        // Save state to storage (thus far, only mappings have been updated, so we must address everything else)
+        _save(cache, /* didChangeBorrowBase: */ false);
+
         // Ensure tokens are transferred
         ERC20 asset_ = asset();
-        cache.lastBalance += amount;
         bool didPrepay = cache.lastBalance <= asset_.balanceOf(address(this));
         if (!didPrepay) {
             asset_.safeTransferFrom(msg.sender, address(this), amount);
         }
```

Comment:
These are the only calls in `deposit` that call outside of the Aloe protocol. In fact, the `safeTransferFrom` call is the only non-view function that even calls outside of this _contract_.

That means it's classified as "interactions" in checks/effects/interactions, so it should go last.